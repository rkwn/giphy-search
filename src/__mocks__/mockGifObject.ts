export const mockGifObject = {
  type: 'gif',
  id: 'cT9lVgPI3ql8AtRDPi',
  url: 'https://giphy.com/gifs/mlb-sports-lets-party-champagne-shower-cT9lVgPI3ql8AtRDPi',
  slug: 'mlb-sports-lets-party-champagne-shower-cT9lVgPI3ql8AtRDPi',
  bitly_gif_url: 'https://gph.is/g/aQbxM32',
  bitly_url: 'https://gph.is/g/aQbxM32',
  embed_url: 'https://giphy.com/embed/cT9lVgPI3ql8AtRDPi',
  username: 'mlb',
  source: 'http://mlb.com',
  title: 'Atlanta Braves Win GIF by MLB',
  rating: 'g',
  content_url: '',
  source_tld: 'mlb.com',
  source_post_url: 'http://mlb.com',
  is_sticker: 0,
  import_datetime: '2021-10-24 04:46:36',
  trending_datetime: '2021-10-24 05:30:06',
  images: {
    original: {
      height: '254',
      width: '480',
      size: '3088100',
      url: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/giphy.gif?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=giphy.gif&ct=g',
      mp4_size: '1239630',
      mp4: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/giphy.mp4?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=giphy.mp4&ct=g',
      webp_size: '854734',
      webp: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/giphy.webp?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=giphy.webp&ct=g',
      frames: '52',
      hash: '127d9ff6f67d78b47ed4cd61687e54b0',
    },
    downsized: {
      height: '254',
      width: '480',
      size: '1750163',
      url: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/giphy-downsized.gif?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=giphy-downsized.gif&ct=g',
    },
    downsized_large: {
      height: '254',
      width: '480',
      size: '3088100',
      url: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/giphy.gif?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=giphy.gif&ct=g',
    },
    downsized_medium: {
      height: '254',
      width: '480',
      size: '3088100',
      url: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/giphy.gif?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=giphy.gif&ct=g',
    },
    downsized_small: {
      height: '100',
      width: '188',
      mp4_size: '151376',
      mp4: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/giphy-downsized-small.mp4?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=giphy-downsized-small.mp4&ct=g',
    },
    downsized_still: {
      height: '254',
      width: '480',
      size: '33711',
      url: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/giphy-downsized_s.gif?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=giphy-downsized_s.gif&ct=g',
    },
    fixed_height: {
      height: '200',
      width: '378',
      size: '1409799',
      url: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/200.gif?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=200.gif&ct=g',
      mp4_size: '802358',
      mp4: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/200.mp4?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=200.mp4&ct=g',
      webp_size: '661318',
      webp: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/200.webp?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=200.webp&ct=g',
    },
    fixed_height_downsampled: {
      height: '200',
      width: '378',
      size: '180075',
      url: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/200_d.gif?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=200_d.gif&ct=g',
      webp_size: '120796',
      webp: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/200_d.webp?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=200_d.webp&ct=g',
    },
    fixed_height_small: {
      height: '100',
      width: '189',
      size: '483539',
      url: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/100.gif?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=100.gif&ct=g',
      mp4_size: '269189',
      mp4: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/100.mp4?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=100.mp4&ct=g',
      webp_size: '246680',
      webp: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/100.webp?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=100.webp&ct=g',
    },
    fixed_height_small_still: {
      height: '100',
      width: '189',
      size: '10477',
      url: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/100_s.gif?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=100_s.gif&ct=g',
    },
    fixed_height_still: {
      height: '200',
      width: '378',
      size: '27094',
      url: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/200_s.gif?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=200_s.gif&ct=g',
    },
    fixed_width: {
      height: '106',
      width: '200',
      size: '566529',
      url: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/200w.gif?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=200w.gif&ct=g',
      mp4_size: '297327',
      mp4: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/200w.mp4?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=200w.mp4&ct=g',
      webp_size: '271604',
      webp: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/200w.webp?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=200w.webp&ct=g',
    },
    fixed_width_downsampled: {
      height: '106',
      width: '200',
      size: '67739',
      url: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/200w_d.gif?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=200w_d.gif&ct=g',
      webp_size: '40960',
      webp: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/200w_d.webp?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=200w_d.webp&ct=g',
    },
    fixed_width_small: {
      height: '53',
      width: '100',
      size: '152603',
      url: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/100w.gif?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=100w.gif&ct=g',
      mp4_size: '49284',
      mp4: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/100w.mp4?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=100w.mp4&ct=g',
      webp_size: '89632',
      webp: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/100w.webp?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=100w.webp&ct=g',
    },
    fixed_width_small_still: {
      height: '53',
      width: '100',
      size: '3766',
      url: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/100w_s.gif?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=100w_s.gif&ct=g',
    },
    fixed_width_still: {
      height: '106',
      width: '200',
      size: '11806',
      url: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/200w_s.gif?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=200w_s.gif&ct=g',
    },
    looping: {
      mp4_size: '4930464',
      mp4: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/giphy-loop.mp4?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=giphy-loop.mp4&ct=g',
    },
    original_still: {
      height: '254',
      width: '480',
      size: '56498',
      url: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/giphy_s.gif?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=giphy_s.gif&ct=g',
    },
    original_mp4: {
      height: '254',
      width: '480',
      mp4_size: '1239630',
      mp4: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/giphy.mp4?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=giphy.mp4&ct=g',
    },
    preview: {
      height: '80',
      width: '151',
      mp4_size: '42043',
      mp4: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/giphy-preview.mp4?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=giphy-preview.mp4&ct=g',
    },
    preview_gif: {
      height: '51',
      width: '96',
      size: '49532',
      url: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/giphy-preview.gif?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=giphy-preview.gif&ct=g',
    },
    preview_webp: {
      height: '84',
      width: '158',
      size: '39258',
      url: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/giphy-preview.webp?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=giphy-preview.webp&ct=g',
    },
    hd: {
      height: '1012',
      width: '1920',
      mp4_size: '7882768',
      mp4: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/giphy-hd.mp4?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=giphy-hd.mp4&ct=g',
    },
    '480w_still': {
      height: '254',
      width: '480',
      size: '3088100',
      url: 'https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/480w_s.jpg?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&rid=480w_s.jpg&ct=g',
    },
  },
  user: {
    avatar_url: 'https://media3.giphy.com/avatars/mlb/UTAk9uV8rZw2.jpg',
    banner_image:
      'https://media3.giphy.com/channel_assets/mlb/8Y39J0Q4HSYL.gif',
    banner_url: 'https://media3.giphy.com/channel_assets/mlb/8Y39J0Q4HSYL.gif',
    profile_url: 'https://giphy.com/mlb/',
    username: 'mlb',
    display_name: 'MLB',
    description:
      'Every GIF from every game of MLB! Find and share all of your favorite moments!',
    instagram_url: 'https://instagram.com/mlb',
    website_url: 'http://mlb.com',
    is_verified: true,
  },
  analytics_response_payload:
    'e=Z2lmX2lkPWNUOWxWZ1BJM3FsOEF0UkRQaSZldmVudF90eXBlPUdJRl9UUkVORElORyZjaWQ9ZGViOGRjOTJ2dDBhdnM2ajFndXlsZDJpeDhma2FxOGRub25nYzkzb21yYmNzdzl4JmN0PWc',
  analytics: {
    onload: {
      url: 'https://giphy-analytics.giphy.com/v2/pingback_simple?analytics_response_payload=e%3DZ2lmX2lkPWNUOWxWZ1BJM3FsOEF0UkRQaSZldmVudF90eXBlPUdJRl9UUkVORElORyZjaWQ9ZGViOGRjOTJ2dDBhdnM2ajFndXlsZDJpeDhma2FxOGRub25nYzkzb21yYmNzdzl4JmN0PWc&action_type=SEEN',
    },
    onclick: {
      url: 'https://giphy-analytics.giphy.com/v2/pingback_simple?analytics_response_payload=e%3DZ2lmX2lkPWNUOWxWZ1BJM3FsOEF0UkRQaSZldmVudF90eXBlPUdJRl9UUkVORElORyZjaWQ9ZGViOGRjOTJ2dDBhdnM2ajFndXlsZDJpeDhma2FxOGRub25nYzkzb21yYmNzdzl4JmN0PWc&action_type=CLICK',
    },
    onsent: {
      url: 'https://giphy-analytics.giphy.com/v2/pingback_simple?analytics_response_payload=e%3DZ2lmX2lkPWNUOWxWZ1BJM3FsOEF0UkRQaSZldmVudF90eXBlPUdJRl9UUkVORElORyZjaWQ9ZGViOGRjOTJ2dDBhdnM2ajFndXlsZDJpeDhma2FxOGRub25nYzkzb21yYmNzdzl4JmN0PWc&action_type=SENT',
    },
  },
};
