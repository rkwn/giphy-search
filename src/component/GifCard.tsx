import { useContext } from 'react';
import { FavoritesContext } from '../context/FavoritesContext';
import { GifObject } from '../api/giphy-types';
import starRegular from '../assets/star-regular.svg';
import starSolid from '../assets/star-solid.svg';

interface GifCardProps {
  gif: GifObject;
}

const GifCard = ({ gif }: GifCardProps) => {
  const { favorites, setFavorites } = useContext(FavoritesContext);

  const isFavorite = favorites?.some((favorite) => favorite.id === gif.id);

  const toggleFavorite = (gif: GifObject) => {
    if (!isFavorite) {
      setFavorites([...favorites, gif]);
    } else {
      setFavorites([...favorites].filter((favorite) => favorite.id !== gif.id));
    }
  };

  return (
    <div className="gif-card" key={gif.id}>
      <img
        className="gif-img"
        src={gif.images.fixed_width_downsampled.url}
        alt="gif"
      />
      <img
        className="gif-favorite-icon"
        src={isFavorite ? starSolid : starRegular}
        alt={'favorite icon'}
        onClick={() => toggleFavorite(gif)}
      />
      <div className="gif-info">{gif.title}</div>
    </div>
  );
};
export default GifCard;
