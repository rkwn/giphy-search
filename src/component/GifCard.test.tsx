import React from 'react';
import { expect } from '@jest/globals';
import { render } from '@testing-library/react';
import { StaticRouter } from 'react-router-dom';
import GifCard from './GifCard';
import { mockGifObject } from '../__mocks__/mockGifObject';

describe('<GifCard />', () => {
  it('displays a gif', () => {
    const toggleFavorite = jest.fn();
    const gifCard = render(
      <StaticRouter>
        <GifCard
          gif={mockGifObject}
          isFavorite={false}
          toggleFavorite={toggleFavorite}
        />
      </StaticRouter>
    );
    const {
      images: {
        fixed_width_downsampled: { url },
      },
    } = mockGifObject;
    const gifImg = gifCard.getByAltText('gif');
    expect(gifImg.src).toContain(url);
  });

  it('should render the correct favorite icon depending on isFavorite prop', () => {
    const toggleFavorite = jest.fn();
    const gifCard = render(
      <StaticRouter>
        <GifCard
          gif={mockGifObject}
          isFavorite={false}
          toggleFavorite={toggleFavorite}
        />
      </StaticRouter>
    );

    const favoriteIcon = gifCard.getByAltText('favorite icon');
    expect(favoriteIcon.src).toContain('star-regular.svg');
  });
});
