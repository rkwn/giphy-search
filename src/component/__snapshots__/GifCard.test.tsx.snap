// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifCard /> GifCard renders correctly 1`] = `"<div><div class=\\"gif-card\\"><img class=\\"gif-img\\" src=\\"https://media1.giphy.com/media/cT9lVgPI3ql8AtRDPi/200w_d.gif?cid=deb8dc92vt0avs6j1guyld2ix8fkaq8dnongc93omrbcsw9x&amp;rid=200w_d.gif&amp;ct=g\\" alt=\\"gif\\"><img class=\\"gif-favorite-icon\\" src=\\"star-regular.svg\\" alt=\\"favorite icon\\"><div class=\\"gif-info\\">Atlanta Braves Win GIF by MLB</div></div></div>"`;
